### ISSUE TITLE
DateOnly type support

### ISSUE THREAD

**Issue by colgreen** [ladeak/JsonMergePatch#66]
Hi,

Our ASP.NET Core Controller projects are targeting .NET 6, and our DTO types make use of System.DateOnly (which is new to .NET 6)

The source generator produces wrapper properties like so for DateOnly typed properties:

```csharp
private System.Nullable<LaDeak.JsonMergePatch.Generated.SafeSystem.DateOnlyWrapped> _dateOfBirth;
public System.Nullable<LaDeak.JsonMergePatch.Generated.SafeSystem.DateOnlyWrapped> DateOfBirth
{
    get { return _dateOfBirth; }
    set
    {
	Properties[11] = true;
	_dateOfBirth = value;
    }
}
```

If I am understanding correctly, I think DateOnly should be detected as a System type and not wrapped. I think that test occurs in `GeneratedTypeFilter`, here:

```csharp
public static bool IsGeneratableType(ITypeSymbol typeInfo)
{
    bool generic = false;
    if (typeInfo is INamedTypeSymbol namedTypeInfo)
	generic = namedTypeInfo.IsGenericType;
    return typeInfo.SpecialType == SpecialType.None && !typeInfo.IsAnonymousType && !typeInfo.IsAbstract && !generic && !typeInfo.IsStatic && typeInfo.TypeKind != TypeKind.Enum;
}
```

The `SpecialType` enum doesn't contain a value for `DateOnly`, because the Microsoft.CodeAnalysis project is targeting .NetStandard 2.0, which doesn't include the `DateOnly` type.

I am not sure there is a nice solution here, but I guess extending that check for new types outside of .NetStandard 2.0 could work?

**Comment by ladeak** [comment]
I think we can consider extending it. You can open a PR or I can take a look in the coming week.

**Comment by ladeak** [comment]
Let me know though if you prefer to have a look.

**Comment by colgreen** [comment]
I'm looking into it. So far it appears there's no easy way to do a strict type check, but I guess using ITypeSymbol.ToDisplayString() might be ok (if just feels a bit hacky).

**Comment by ladeak** [comment]
Released with v6.1.0

**Comment by colgreen** [comment]
Thank you.
